### PR TITLE
PS-8337: Remove keyring encryption

### PR DESCRIPTION
**Pull request by dutow**
This is a single commit that removes the old keyring encryption code.

The  linear version of this change can be found at https://github.com/dutow/percona-server/tree/ps-8.0-enc-removal31

### CODE REVIEW

File: sql/log_event.cc [hunk 1]
```diff
@@ -4909,19 +4909,29 @@ int Query_log_event::do_apply_event(Relay_log_info const *rli,
         }
 
         if (sqlcom_can_generate_row_events(thd->lex->sql_command) &&
-            thd->get_row_count_func() > 0 &&
```
Comment (oleksandr-kachan) [inline sql/log_event.cc]: @dutow, just to recheck, is this change expected to be here? This looks like one of the conflicts in 8.0.31 merge.

Comment (dutow) [reply sql/log_event.cc]: No, this and the related change in sql/sql_class shouldn't be there, good catch, I'll revert this.

File: sql/sql_base.cc [hunk 2]
```diff
@@ -6815,8 +6815,6 @@ bool open_tables_for_query(THD *thd, TABLE_LIST *tables, uint flags) {
 
   if (open_secondary_engine_tables(thd, flags)) goto end;
 
-  thd->check_rpl_stmt_event_format_used();
```
Comment (oleksandr-kachan) [inline sql/sql_base.cc]: AFAIR this line is related to merge conflicts as well.

File: sql/sql_class.cc [hunk 3]
```diff
@@ -718,8 +718,7 @@ THD::THD(bool enable_plugins)
       m_is_plugin_fake_ddl(false),
       m_inside_system_variable_global_update(false),
       bind_parameter_values(nullptr),
-      bind_parameter_values_count(0),
-      is_rpl_stmt_event_format_used(true) {
+      bind_parameter_values_count(0) {
```
Comment (oleksandr-kachan) [inline sql/sql_class.cc]: Same as above and also some methods removed below.

Comment (dutow) [reply sql/sql_class.cc]: Yes, I fixed all occurrences of is_rpl_stmt_event_format_used.

File: sql/log_event.cc (hunk 1 above)
Comment (oleksandr-kachan) [reply sql/log_event.cc]: Thanks!

File: storage/federated/ha_federated.cc [hunk 4]
```diff
@@ -2990,13 +2990,28 @@ int ha_federated::real_connect() {
...
+      result in the case of timeout exceeding. See PS-7999 for details.
+    */
+    if (!mysql || (mysql_errno(mysql) != CR_SERVER_LOST &&
+                   mysql->net.error != NET_ERROR_SOCKET_UNUSABLE))
+      goto end;
+    thd->get_stmt_da()->reset_diagnostics_area();
+    thd->get_stmt_da()->reset_condition_info(thd);
+    if (!mysql->reconnect || mysql_reconnect(mysql)) goto end;
+    rc = mysql_real_query(mysql, query, len);
+  }
```
Comment (percona-ysorokin) [inline storage/federated/ha_federated.cc]: This looks like something from a different patch

Comment (dutow) [reply storage/federated/ha_federated.cc]: Yes, that's another .31 merge change I missed, fixed.

File: storage/innobase/fil/fil0fil.cc [hunk 5]
```diff
@@ -8332,8 +7923,9 @@ dberr_t Fil_shard::do_io(const IORequest &type, bool sync,
   }
 
   if (page_size.is_compressed()) {
-    req_type.mark_page_zip_compressed();
-    req_type.set_zip_page_physical_size(page_size.physical());
+    // TODO
+    //req_type.mark_page_zip_compressed();
+    //req_type.set_zip_page_physical_size(page_size.physical());
```
Comment (oleksandr-kachan) [inline storage/innobase/fil/fil0fil.cc]: @dutow, is it expected to stay like this for now?

File: storage/innobase/fil/fil0fil.cc [hunk 6]
```diff
@@ -8863,8 +8451,8 @@ static dberr_t fil_iterate(const Fil_page_iterator &iter, buf_block_t *block,
       ut_d(block->page.zip.m_external = true);
       ut_ad(iter.m_page_size == callback.get_page_size().physical());
 
-      read_request.mark_page_zip_compressed();
-      read_request.set_zip_page_physical_size(iter.m_page_size);
+      //read_request.mark_page_zip_compressed();
+      //TODO read_request.set_zip_page_physical_size(iter.m_page_size);
```
Comment (oleksandr-kachan) [inline storage/innobase/fil/fil0fil.cc]: Same here

File: storage/innobase/handler/ha_innodb.cc [hunk 7]
```diff
@@ -6199,6 +5997,7 @@ static int innobase_init_files(dict_init_mode_t dict_init_mode,
     dd_space.add_file(&dd_file);
     tablespaces->push_back(&dd_space);
 
+     // we were maybe only missing this?
```
Comment (oleksandr-kachan) [inline storage/innobase/handler/ha_innodb.cc]: Is this something that requires additional attention at this stage?

File: storage/innobase/handler/ha_innodb.cc [hunk 8]
```diff
@@ -8047,38 +7846,41 @@ int ha_innobase::open(const char *name, int, uint open_flags,
...
+  // my_error(ER_CANNOT_FIND_KEY_IN_KEYRING, MYF(0));
+  // dict_table_close(ib_table, FALSE, FALSE);
+  // ib_table = NULL;
+  // is_part = NULL;
+  // free_share(m_share);
+
+  // DBUG_RETURN(ret_err);
+  //}
+  //}
+  //}
```
Comment (oleksandr-kachan) [inline storage/innobase/handler/ha_innodb.cc]: Should this be removed?

File: storage/innobase/fil/fil0fil.cc (hunk 5 above)
Comment (dutow) [reply storage/innobase/fil/fil0fil.cc]: We don't need these, looks like I lost this change when I rebased to .31, I'll remove these lines.

File: storage/innobase/fil/fil0fil.cc (hunk 6 above)
Comment (dutow) [reply storage/innobase/fil/fil0fil.cc]: Same, this was related to keyring encryption.

File: storage/innobase/handler/ha_innodb.cc (hunk 7 above)
Comment (dutow) [reply storage/innobase/handler/ha_innodb.cc]: Same, leftover comment.

File: storage/innobase/handler/ha_innodb.cc (hunk 8 above)
Comment (dutow) [reply storage/innobase/handler/ha_innodb.cc]: Yes, removed.

File: storage/innobase/fil/fil0fil.cc [hunk 9]
```diff
@@ -3425,17 +3313,15 @@ fil_space_t *Fil_shard::space_create(const char *name, space_id_t space_id,
 
   space->is_corrupt = false;
 
-  space->is_space_encrypted = false;
-
-  space->exclude_from_rotation = false;
+  //space->is_space_encrypted = false;
```
Comment (oleksandr-kachan) [inline storage/innobase/fil/fil0fil.cc]: Sorry, one more question to this file. Can this be removed as well?

Comment (dutow) [reply storage/innobase/fil/fil0fil.cc]: yes, done.

File: unittest/gunit/CMakeLists.txt [hunk 10]
```diff
@@ -315,10 +315,6 @@ IF(MY_COMPILER_IS_GNU AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12)
   ADD_COMPILE_FLAGS(regexp_engine-t.cc COMPILE_FLAGS "-Wno-restrict")
 ENDIF()
 
-IF(MY_COMPILER_IS_GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8)
-  ADD_COMPILE_FLAGS(hypergraph_optimizer-t.cc connect_joins-t.cc COMPILE_FLAGS "-Wno-unused-variable")
-ENDIF()
-
```
Comment (oleksandr-kachan) [inline unittest/gunit/CMakeLists.txt]: Is this related to encryption?

File: unittest/gunit/keyring/CMakeLists.txt [hunk 11]
```diff
@@ -56,7 +56,6 @@ LIST(APPEND SRC_FILES
   ${CMAKE_SOURCE_DIR}/plugin/keyring/checker/checker_ver_2_0.cc
   ${CMAKE_SOURCE_DIR}/unittest/gunit/keyring/buffered_file_io_10.cc
   ${CMAKE_SOURCE_DIR}/unittest/gunit/keyring/psi_memory_key.cc
-  ${CMAKE_SOURCE_DIR}/unittest/gunit/keyring/uuid.cc
```
Comment (oleksandr-kachan) [inline unittest/gunit/keyring/CMakeLists.txt:59]: Cannot find if gunit/keyring/uuid.cc was removed as well?

Comment (dutow) [reply unittest/gunit/keyring/CMakeLists.txt:59]: Yes, the keyname uuid support was related to keyring encryption.

Comment (oleksandr-kachan) [reply unittest/gunit/keyring/CMakeLists.txt:59]: I don't see this file being deleted in the change. Should we delete it?

File: unittest/gunit/CMakeLists.txt (hunk 10 above)
Comment (dutow) [reply unittest/gunit/CMakeLists.txt]: No, it's unrelated, but looks like it didn't cause any build issues for me and I didn't notice it. I'll fix it.

File: unittest/gunit/keyring/CMakeLists.txt (hunk 11 above)
Comment (dutow) [reply unittest/gunit/keyring/CMakeLists.txt:59]: Yes, it is deleted in the linear branch version, but I missed the removal in this single commit PR. I'll delete it.

Comment (oleksandr-kachan) [reply unittest/gunit/keyring/CMakeLists.txt:59]: Thanks @dutow

**Comment by oleksandr-kachan** [comment]
Hi @dutow, rebase this change please, there are a few conflicts.

File: mysys/system_key.cc [hunk 12]
```diff
@@ -64,7 +64,7 @@ uchar *parse_system_key(const uchar *key, const size_t key_length,
 
   if (key == nullptr || key_length == 0) return nullptr;
 
-  for (; key_version_length < key_length && key[key_version_length] != ':';
+  for (; key[key_version_length] != ':' && key_version_length < key_length;
```
Comment (percona-ysorokin) [inline mysys/system_key.cc]: Any reason for swapping these conditions?

**Review by percona-ysorokin — APPROVED** [review]
LGTM with single question

File: mysys/system_key.cc (hunk 12 above)
Comment (dutow) [reply mysys/system_key.cc]: It was part of one of the big keyring encryption commits. But you are right, the current version is better, I'll revert this change.

**Review by oleksandr-kachan — APPROVED** [review]

File: storage/innobase/dict/dict0dd.cc [hunk 13]
```diff
@@ -7255,10 +7231,6 @@ bool dd_clear_encryption_flag(THD *thd, const char *space_name,
     dd_space->se_private_data().set(dd_space_key_strings[DD_SPACE_FLAGS],
                                     dd_space_flags);
     dd_space->options().set("encryption", "N");
-    if (clear_online_encryption) {
```
Comment (oleksandr-kachan) [inline storage/innobase/dict/dict0dd.cc:7258]: @dutow, looks like  clear_online_encryption should be removed from this method

`/home/vsts/work/1/s/storage/innobase/dict/dict0dd.cc:7224:23: error: lambda capture 'clear_online_encryption' is not used [-Werror,-Wunused-lambda-capture]
  auto update_func = [clear_online_encryption](dd::Tablespace *dd_space) {
                      ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
`

There may be some other Azure compile issues, haven't checked them all yet.

File: unittest/gunit/keyring/keyring-api-t.cc [hunk 14]
```diff
@@ -364,82 +321,81 @@ TEST_F(Keyring_api_test, GeneratePBRotatePBFetchFirstVersionFetchLatestPB) {
...
-  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver2), "2:", 2) == 0);
+  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver1), "1:", 2) == 0);
   my_free(key_type);
   key_type = NULL;
 
   // make sure that rotated key is different than the original one
-  // + 2 to skip key version of retrieved latest percona_binlog key
-  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver1),
-                     reinterpret_cast<char *>(key_ver2) + 2, 16) != 0);
+  ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver0) + 2,
```
Comment (oleksandr-kachan) [inline unittest/gunit/keyring/keyring-api-t.cc]: @dutow, please consider changing this line to 

ASSERT_TRUE(memcmp(reinterpret_cast<char *>(key_ver0),

ASAN complains here

`==342140==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6040000126c0 at pc 0x7f33f46abf50 bp 0x7ffd4a052b90 sp 0x7ffd4a052340

READ of size 16 at 0x6040000126c0 thread T0
#0 0x7f33f46abf4f in MemcmpInterceptorCommon(void*, int (*)(void const*, void const*, unsigned long), void const*, void const*, unsigned long) (/lib64/libasan.so.8+0xabf4f)
#1 0x7f33f46ac5e8 in memcmp (/lib64/libasan.so.8+0xac5e8)
#2 0x6335ee in keyring__api_unittest::Keyring_api_test_GeneratePBRotatePBFetchFirstVersionFetchLatestPB_Test::TestBody() unittest/gunit/keyring/keyring-api-t.cc:346`

File: unittest/gunit/keyring/keys_container-t.cc [hunk 15]
```diff
@@ -1936,14 +1936,14 @@ TEST_F(Keys_container_with_mocked_system_keys_container_test,
   IKeyring_io *keyring_io = new Buffered_file_io(logger);
   Mock_system_keys_container *system_keys_container =
       new Mock_system_keys_container;
+  keys_container->set_system_keys_container(system_keys_container);
```
Comment (oleksandr-kachan) [inline unittest/gunit/keyring/keys_container-t.cc]: Looks like change for this line should be reverted. At the moment test fails, also there is ASAN error 

==436521==ERROR: AddressSanitizer: heap-use-after-free on address 0x613000005500 at pc 0x0000007188fa bp 0x7fff52e85bb0 sp 0x7fff52e85ba8
#2 0x62143d in keyring__keys_container_unittest::Keys_container_with_mocked_system_keys_container_test_ErrorFromRotateKeyWhenStoringKey_Test::TestBody() unittest/gunit/keyring/keys_container-t.cc:1943

freed by thread T0 here:
...
#9 0x62113f in keyring__keys_container_unittest::Keys_container_with_mocked_system_keys_container_test_ErrorFromRotateKeyWhenStoringKey_Test::TestBody() unittest/gunit/keyring/keys_container-t.cc:1940

**Pull request merged**